sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -595,10 +607,7 @@ case class JsonToStructs(
     if (json.toString.trim.isEmpty) return null
 
     try {
-      converter(parser.parse(
-        json.asInstanceOf[UTF8String],
-        CreateJacksonParser.utf8String,
-        identity[UTF8String]))
+      converter(parser.parse(json.asInstanceOf[UTF8String]))
     } catch {
       case _: BadRecordException => null
```

Comment:
Do we still need this catching of `BadRecordException`?

Reply:
`BadRecordException` shouldn't propagate here any more. I will remove the exception handler and re-run tests.